cmd/opampsupervisor/specification/README.md

```diff
@@ -473,6 +477,149 @@ to allow this.
+                ContentHash: agentSHA256Hash,
+                Signature:   signature,
+            },
+        },
+    },
+    AllPackagesHash: allPackagesHash,
+}
+```
+
+
+### Signing
+
+In order for an agent package to be accepted by the supervisor, the
+package MUST be signed. The supervisor expects the agent tarball to be signed
+using [Cosign], using the keyless signing method to generate separate certificate 
+and signature files.
+
+Both the certificate and cosign signature will be used to create the
+file's signature in the PackagesAvailable message.
+
+During release, the [opentelemetry-collector-releases](https://github.com/open-telemetry/opentelemetry-collector-releases)
+repository uses Cosign to sign the artifacts using a OpenID Connect identity token.  This token is used to request a certificate from
+[Fulcio].
+This certificate contains the identity used to sign the certificate,
+and can be used to verify the signer.
+
+After Fulcio verifies the token, it signs a short-lived x509 certificate for 
+the release workflow. This certificate's private key is used to compute
+a signature for the release artifact (in this case, the agent tarball).
+
+Afterwards, the certificate and signature are written to 
+[Rekor], a transparency log.
+
+As a result, both the certificate and signature together can be used to verify
+the identity of the signer, verify that the artifact that was downloaded
+is the same artifact that was signed, and verify the certificate/signature
+was not tampered with via its inclusion in Rekor.
+
+For more information on keyless signing and Cosign, see the
+[Cosign docs](https://docs.sigstore.dev/cosign/signing/overview/).
+
+### Signature Format
+
+In the PackagesAvailable message, the signature MUST be created
+by joining the base64 encoded certificate and signature files generated 
+by Cosign with a space. Note that the files generated by Cosign are already 
+base64 encoded and do not need to be encoded further.
+
+The signature format will therefore be like this:
+```
+${b64_certificate} ${b64_signature}
+```
+
+### Signature Verification
+
+The signature is verified using Cosign. In order to verify the signature,
+serveral things are needed:
+* A set of identities (issuer/subject pairs) that the signature must
+signed by.
+  * By default, this will be issuer/subject used to sign the agent release
+  from the [opentelemetry-collector-releases](https://github.com/open-telemetry/opentelemetry-collector-releases)
+  repository.
+* (optional) the name of the github repository that generated the artifact.
+  * By default, this is "open-telemetry/opentelemetry-collector-releases".
+  * This field may be explicitly set empty to skip verifying the repository field
+  in the certificate
+* A set of [Fulcio] root certificates.
+  * These certificates are retrieved from "https://tuf-repo-cdn.sigstore.dev"
```

Comment:
It would be interesting to explore the possibility of offline certificate verification. We could embed the root certificates in the Supervisor at build time, but I don't know if the verification process itself can be done without connecting to Rekor.
This can be a future work.

Reply:
there is an option to skip the Rekor/transparency log verification. I couldn't get it to work as I was developing, but I figure it's probably something simple I was missing. We should definitely be able to get it working in some follow-up.